### ISSUE TITLE
Resolving the main app via DNS from an accessory container

### ISSUE THREAD

**Issue by juliusv** [basecamp/kamal#1278]
I'm not sure if this is a feature request or is already possible somehow: I want to reach my main Kamal app from an accessory container (in my specific case, the accessory container is a Prometheus server scraping the /metrics endpoint of the main app). However, I've been unable to find any stable DNS name for the main app, since the container name changes each time the app is deployed, and even that one is not reachable. I was only able to resolve something like `168.146.123.140-53f792ef1caf`, which is listed as the `Hostname` in `docker inspect`.

It would be nice to make the main app resolvable under a predictable and stable DNS name from the other containers, like it's currently possible to resolve accessories from the main app.

**Comment by juliusv** [comment]
@frenkel Thanks, that worked for me :) (just that `host:` from the example needs to be `hosts:` now and expects a list)

**Comment by frenkel** [comment]
Great! Maybe you should close this issue then?

**Comment by juliusv** [comment]
> Maybe you should close this issue then?

I was wondering that as well, it depends on how hacky we judge the linked solution to be I'd say :) Maybe it'd still be nice to support a stable hostname for the app by default, or at least make it a more explicit Kamal setting than having to pass in custom Docker options?

**Comment by viktorianer** [comment]
> Maybe it'd still be nice to support a stable hostname for the app by default

I would prefer this solution and should be easy to do, as it needs only a name passed to the Docker commands?!

**Comment by juliusv** [comment]
I agree it's very easy once you know about it, but it's not documented and I would not otherwise have easily figured it out.

I think since this is going to be a common need it should be at least documented then? But even better if there was just a stable name by default.

**Comment by viktorianer** [comment]
> But even better if there was just a stable name by default.

I think you misunderstood my comment. No one should have to search the README to solve this issue (not easy to identify).

My point was precisely this: all containers should have a default name, and implementing that should be straightforward.

PS: Before the new year, I won’t have any time to look into this. Additionally, I have several other open tasks in the Rails repository that I’d like to start/finish first. 🤷🏼‍♂️

**Comment by juliusv** [comment]
@viktorianer Sorry, I misread you indeed! Yes, that sounds good :)

**Comment by juliusv** [comment]
@djmb I see,  though now I'm running into other situations where I need an accessory to be able to talk back to my main app (the `zero-cache` component from https://zero.rocicorp.dev/ needs to be able to call back into my web app to run some custom mutator endpoints).

If a stable identity for each container is not possible due to the issues you mentioned, maybe what I'm actually looking for is just a load-balanced endpoint that works the same as the external proxy endpoint for the app. Or is that already reachable somehow from within the container?

**Comment by djmb** [comment]
@juliusv - I think you could maybe use something like in the accessory config:

```
    options:
      add-host: "<app-host-name>:host-gateway"
```

Then you should be able to connect directly to the host via `https://<app-host-name>/`.

**Comment by juliusv** [comment]
@djmb Thanks! Hmm, I haven't gotten that to work quite yet. The accessory experiences a connection timeout on port 443 when trying to connect to the app that way. As `<app-host-name>` I used the same domain name as the `host` parameter for the main app in my Kamal config.

For now what's working ok enough for me is the `alias` trick from earlier in this issue, although I realize that might introduce intermittent issues whenever I deploy a new version of the app.
